cloud_function/main.py

```diff
@@ -0,0 +1,127 @@
+import os
+import json
+import requests
+from google.cloud import storage
+from google.cloud import exceptions
+
+project = os.environ.get("GCP_PROJECT", "broad-gotc-dev")
+wfl_url = os.environ.get("WFL_URL", "https://workflow-launcher.gotc-dev.broadinstitute.org")
+default_credentials = os.environ.get("DEFAULT_CREDENTIALS", False)
+
+
+def get_test_credentials():
+    import google.auth
+    import google.auth.transport.requests
+
+    scopes = ["https://www.googleapis.com/auth/cloud-platform",
+              "https://www.googleapis.com/auth/userinfo.email",
+              "https://www.googleapis.com/auth/userinfo.profile"]
+    credentials, gcp_project = google.auth.default(scopes=scopes)
+    if not credentials.valid:
+        credentials.refresh(google.auth.transport.requests.Request())
+    return credentials
+
+def get_auth_headers():
+    scopes_list = ["https://www.googleapis.com/auth/cloud-platform",
+                   "https://www.googleapis.com/auth/userinfo.email",
+                   "https://www.googleapis.com/auth/userinfo.profile"]
+    scopes = ",".join(scopes_list)
+    metadata_url = f'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes={scopes}'
+    metadata_headers = {'Metadata-Flavor': 'Google'}
+    r = requests.get(metadata_url, headers=metadata_headers)
+    r.raise_for_status()
+    access_token = r.json()['access_token']
+    headers = {
+        'Authorization': 'Bearer {}'.format(access_token)
+    }
+    return headers
+
+def get_manifest_path(object_name):
+    parts = object_name.split("/")
+    chipwell_barcode = parts[0]
+    analysis_version = parts[1]
+    return "/".join([chipwell_barcode, analysis_version, "ptc.json"])
+
+def get_or_create_workload(headers, cromwell_url):
+    payload = {
+        "creator": "wfl-non-prod@broad-gotc-dev.iam.gserviceaccount.com",
```

Comment:
I assume once the deploy script of this function supports prod, this payload will also needs to be parameterized? (use different creator values for dev and prod?)

Reply:
Ah good catch! Turns out the cloud function VM comes with the "FUNCTION_IDENTITY" env variable set to the service account email so we don't have to pass it in